provider/webhook/webhook.go

```diff
@@ -270,11 +279,15 @@ func (p WebhookProvider) AdjustEndpoints(e []*endpoint.Endpoint) ([]*endpoint.En
 	if resp.StatusCode != http.StatusOK {
 		adjustEndpointsErrorsGauge.Inc()
 		log.Debugf("Failed to AdjustEndpoints with code %d", resp.StatusCode)
-		return nil, fmt.Errorf("failed to AdjustEndpoints with code %d", resp.StatusCode)
+		err := fmt.Errorf("failed to AdjustEndpoints with code  %d", resp.StatusCode)
+		if isRetryableError(resp.StatusCode) {
+			return nil, provider.NewSoftError(err)
+		}
+		return nil, err
 	}
 
 	if err := json.NewDecoder(resp.Body).Decode(&endpoints); err != nil {
-		recordsErrorsGauge.Inc()
+		adjustEndpointsErrorsGauge.Inc()
```

Comment:
this seems to be an unrelated fix, maybe better to create a separate PR

Reply:
@SimonKienzler ^

Reply:
I thought it might pass as boyscouting, will open a separate PR for it.

Reply:
Reverted the change and will open a separate PR for the fix soon. PTAL

Reply:
See #4374